### PR TITLE
read all bytes available from wire protocol in MConnection test

### PR DESCRIPTION
**Pull request by petabytestorage**

### CODE REVIEW

**Comment by ebuchman** [comment]
hmm what's the story here?

**Comment by petabytestorage** [comment]
reading the source code, i noticed that it seems like the channel would contain the wire-encoded message not the literal message bytes so it was short some bytes for reading on the other end of any ReadWriteCloser. Through guess and check i found it seemed like 4 bytes was the extra space for reasons i am not sure yet @ebuchman however it seemed odd enough to deserve a PR. when i use 5 instead of 4 my test fails so from that i infer i've guessed the right padding.

**Comment by petabytestorage** [comment]
i think this explains the length written: https://github.com/tendermint/tendermint/blob/develop/p2p/connection.go#L663

File: p2p/connection_test.go [hunk 1]
```diff
@@ -44,12 +44,12 @@ func TestMConnectionSend(t *testing.T) {
 	assert.True(mconn.Send(0x01, msg))
 	// Note: subsequent Send/TrySend calls could pass because we are reading from
 	// the send queue in a separate goroutine.
-	server.Read(make([]byte, len(msg)))
+	server.Read(make([]byte, len(wire.BinaryBytes(msg))+4))
```
Comment (melekes) [inline p2p/connection_test.go]: have you figured out where 4 bytes are coming from? 1 byte is packetMsgType

**Comment by ebuchman** [comment]
Should I close this for now ?

**Pull request closed without merging**